addons/point_of_sale/models/pos_session.py

```diff
@@ -744,8 +744,11 @@ def _accumulate_amounts(self, data):
                         # for taxes
                         tuple((tax['id'], tax['account_id'], tax['tax_repartition_line_id']) for tax in line['taxes']),
                         line['base_tags'],
+                        # closing entry by product
+                        line['product_id']
                     )
```

Comment:
I would have change the sale_key used to group, rather than changing how the key is applied. Something along the line of:
`line['product_id'] --> is_closing_entry_by_product and line['product_id'] or True`

That way, if you change they key, I think you don't need to change _update_amounts by _update_quantities

Reply:
@zepa-odoo This is actually a good insight. Kindly consider this comment.

Reply:
@fpodoo it's already check in _prepare_line
`'product_id': order_line.product_id.id if self.config_id.is_closing_entry_by_product else False`